stdlib/include/qir_stdlib.h

```diff
@@ -0,0 +1,243 @@
+
+    // Adds the given integer value to the reference count for the big integer. Deallocates the big integer if the
+    // reference count becomes 0. The behavior is undefined if the reference count becomes negative.
+    QIR_SHARED_API void __quantum__rt__bigint_update_reference_count(QirBigInt*, int32_t); // NOLINT
+
+    // Returns the negative of the big integer.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_negate(QirBigInt*); // NOLINT
+
+    // Adds two big integers and returns their sum.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_add(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Subtracts the second big integer from the first and returns their difference.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_subtract(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Multiplies two big integers and returns their product.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_multiply(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Divides the first big integer by the second and returns their quotient.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_divide(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Returns the first big integer modulo the second.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_modulus(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Returns the big integer raised to the integer power.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_power(QirBigInt*, int32_t); // NOLINT
+
+    // Returns the bitwise-AND of two big integers.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_bitand(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Returns the bitwise-OR of two big integers.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_bitor(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Returns the bitwise-XOR of two big integers.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_bitxor(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Returns the bitwise complement of the big integer.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_bitnot(QirBigInt*); // NOLINT
+
+    // Returns the big integer arithmetically shifted left by the (positive) integer amount of bits.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_shiftleft(QirBigInt*, int64_t); // NOLINT
+
+    // Returns the big integer arithmetically shifted right by the (positive) integer amount of bits.
+    QIR_SHARED_API QirBigInt* __quantum__rt__bigint_shiftright(QirBigInt*, int64_t); // NOLINT
+
+    // Returns true if the two big integers are equal, false otherwise.
+    QIR_SHARED_API bool __quantum__rt__bigint_equal(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Returns true if the first big integer is greater than the second, false otherwise.
+    QIR_SHARED_API bool __quantum__rt__bigint_greater(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Returns true if the first big integer is greater than or equal to the second, false otherwise.
+    QIR_SHARED_API bool __quantum__rt__bigint_greater_eq(QirBigInt*, QirBigInt*); // NOLINT
+
+    // Q# Math:
```

Comment:
It probably makes sense to update this comment to something not-Q# specific now. The status of these math functions is still a little TBD (should they be updated to be `__rt__`, removed in favor of generated functions that use llvm intrinsics, etc).

Reply:
@swernli  For this PR I'm trying to move it over with as few changes as possible. Then we can update the lib.